src/Avalonia.Visuals/Rendering/ImmediateRenderer.cs

```diff
@@ -224,6 +224,9 @@ private void Render(DrawingContext context, IVisual visual, Rect clipRect)
             var clipToBounds = visual.ClipToBounds;
             var bounds = new Rect(visual.Bounds.Size);
 
+            if (visual == _root)
+                context.PlatformImpl.Clear(Colors.Transparent);
```

Comment:
This method is called for every visual in the window, so i don't think this is the right place to be clearing the window!